applications/OptimizationApplication/custom_python/add_custom_response_utilities_to_python.cpp

```diff
@@ -0,0 +1,43 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   license: OptimizationApplication/license.txt
+//
+//  Main author:     Suneth Warnakulasuriya
+//
+
+// System includes
+
+// External includes
+
+// Project includes
+
+// Application includes
+#include "custom_utilities/response/mass_response_utils.h"
+
+// Include base h
+#include "add_custom_response_utilities_to_python.h"
+
+namespace Kratos {
+namespace Python {
+
+void  AddCustomResponseUtilitiesToPython(pybind11::module& m)
+{
+    namespace py = pybind11;
+
+    py::class_<MassResponseUtils >(m, "MassResponseUtils")
+        .def_static("CalculateMass", &MassResponseUtils::CalculateMass)
```

Comment:
I will move the if blocks to cpp side

Reply:
And will remove the old implementation, and re-use the new implementation.

Reply:
Ok, I am sorry, but the exisitng implementation cannot be removed because of the following reasons
1. The existing implementation uses projected values to compute mass, and its derivatives. This is not done in the new deisgn. `Responses` does not need to know whether their `THICKNESS`, `DENSITY`, `CROSS_AREA` are projected or not. They should be able to calculate mass sensitivity w.r.t. each `THICKNESS`, `DENSITY`, `CROSS_AREA`, etc without ever knowing whether they are independent/projected or not. The control mechanism decides whether they are independent or dependent, based on that control should have modification within it self. Otherwise, we will need seperate mass response for different control mechanisms such as SIMP, raw mass response for SI, raw mass response for constraints which is unmanagable.
2. If I am to use the proposed `MassResponseUtils` inside exsiting responses, the senstivitiy values will be wrong, hence the tests will be useless.

Therefore, we have to live with both implementations for the time being untill the new control mechanism which works with this proposed new `Response` design is implemented to properly port these tests. I found this same reason when I was looking at this the first time, that was why I could not replace existing reposnes with the new ones. But I forgot the reasons. I wrote it here for my reference as well.